File: app/src/main/java/uk/co/ourfriendirony/medianotifier/clients/MovieClient.java
package uk.co.ourfriendirony.medianotifier.clients;

import java.io.IOException;
import java.util.List;

import uk.co.ourfriendirony.medianotifier.clients.tmdb.TMDBClient;
import uk.co.ourfriendirony.medianotifier.mediaitem.MediaItem;

public class MovieClient implements Client {

    @Override
    public List<MediaItem> searchMediaItem(String name) throws IOException {
        return new TMDBClient().queryMovie(name);
    }

    @Override
    public MediaItem getMediaItem(String id) throws IOException {
        return new TMDBClient().getMovie(id);
    }
}

File: app/src/main/java/uk/co/ourfriendirony/medianotifier/clients/MovieClient.kt
package uk.co.ourfriendirony.medianotifier.clients

import uk.co.ourfriendirony.medianotifier.clients.tmdb.TMDBClient
import uk.co.ourfriendirony.medianotifier.mediaitem.MediaItem
import java.io.IOException

class MovieClient : Client {
    @Throws(IOException::class)
    override fun searchMediaItem(name: String?): List<MediaItem?> {
        return TMDBClient().queryMovie(name)
    }

    @Throws(IOException::class)
    override fun getMediaItem(id: String?): MediaItem {
        return TMDBClient().getMovie(id)
    }
}